apps/portal/theme/templates/dashboard.jag

```diff
@@ -31,22 +31,42 @@
         </div>
         <div class="collapse navbar-collapse horizontal-bar">
             <ul class="nav navbar-nav">
-                <% if(dashboard.pages.length >= 2){  %>
-                <li<% if(dashboard.landing === pageId) { %> class="active"<% } %>>
-                    <a href="<%=dbRoot%>">Home</a>
-                </li>
+                <% if(dashboard.pages.length >= 2) { %>
+                    <li
+                        <% if(dashboard.landing === pageId) { %>
+                            class="active"
+                        <% } %>>
+                            <a href="<%=dbRoot%>">Home</a>
+                    </li>
+                <% } %>
+
                 <%
-}
+                var usr = require('/modules/user.js');
+                var user = usr.current();
+                var pages = dashboard.pages;
+                var isAnonView = JSON.parse(request.getParameter('isAnonView'));
+
+                pages.forEach(function (p) {
+                    if(isAnonView || !user) {
+                        if (p.isanon) {
```

Comment:
I changed p as a page. Since isanon is a dashboard json file property, If we are going to change that it involves some major code refactoring. In all the places in UES, we call it as isanon and that is the way it defined in a dashboard json file.